### PR TITLE
Add a WebGL 2 test for array length expressions that have side effects

### PR DESCRIPTION
**Pull request by Oletus**
ESSL3 defines expressions in section 5.9 so that the length() method is
only allowed to be called on array names, not on complex expressions that
return arrays. Enforce this restriction, which simplifies constant
folding of length() method calls.

### CODE REVIEW

File: sdk/tests/conformance2/glsl3/array-length-side-effects.html [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+out vec4 my_FragColor;
+
+void main() {
+    int a = (int[1](0)).length();
+    my_FragColor = vec4(float(a));
+}
+</script>
+<script type="text/javascript">
+"use strict";
+description();
```
Comment (zhenyao) [inline sdk/tests/conformance2/glsl3/array-length-side-effects.html]: Can you add a comment here, pointing to the text in the spec leading to this test? Otherwise people will very likely keep asking questions because this is a corner case and is different from Desktop GL.

**Comment by Oletus** [comment]
zhenyao: I updated the pull request with the comment.

**Comment by zhenyao** [comment]
Thanks.  LGTM.
